src/components/Amount.vue

```diff
@@ -1,25 +1,38 @@
 <template>
     <span class="amount" :class="{ approx: showApprox && isApprox }">
         {{ formattedAmount }}
-        <span class="nim">NIM</span>
+        <span class="currency" :class="currency">{{currency}}</span>
     </span>
 </template>
 
 <script lang="ts">
-import { Component, Prop, Vue } from 'vue-property-decorator';
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+import { FormattableNumber } from '@nimiq/utils';
+type BigInteger = import ('big-integer').BigInteger;
 
 @Component
 export default class Amount extends Vue {
-    // Amount in luna
-    @Prop(Number) public amount!: number;
+    // Amount in smallest unit
+    @Prop({type: [Number, Object]}) public amount!: number | BigInteger;
     // If set takes precedence over minDecimals and maxDecimals
-    @Prop({type: Number, validator: Amount._validateDecimals }) public decimals?: number;
-    @Prop({type: Number, default: 2, validator: Amount._validateDecimals }) public minDecimals!: number;
-    @Prop({type: Number, default: 5, validator: Amount._validateDecimals }) public maxDecimals!: number;
+    @Prop(Number) public decimals?: number;
+    @Prop({type: Number, default: 2}) public minDecimals!: number;
+    @Prop({type: Number, default: 5}) public maxDecimals!: number;
+    @Prop({type: Number, default: 5}) public totalDecimals!: number;
```

Comment:
So this is simply confusing now. What are `decimals` for, and what are `totalDecimals` in comparison to `minDecimals` and `maxDecimals`?

Looking through the code it seems that `totalDecimals` has nothing to do with the displayed decimals (but the three other *Decimals do), but is only relevant to convert from "coin" to smallest unit. So I think this is better named someting along `unitToCoinRatio`, which would then be `100000` for NIM and you can use log10() for moving the decimal separator:
```ts
FormattableNumber.moveDecimalSeparator(-Math.log10(this.totalDecimals))
```
I think naming the prop different because it has a different function makes sense.

Reply:
I like the suggestion as it not only clearly separates the different functions of the props, but also allows for non power of ten ratios, as there in fact are currencies which are not using powers of ten. There are not a lot of them, but they do exist. Daniel liked it better with a more uniform naming, though.

`decimals` has always been redundant, as a shorthand for `minDecimals = maxDecimals = decimals`.